converter_test.go

```diff
@@ -1135,6 +1137,27 @@ func TestArrowToRecord(t *testing.T) {
 			builder:  array.NewStringBuilder(pool),
 			append:   func(b array.Builder, vs interface{}) { b.(*array.StringBuilder).AppendValues(vs.([]string), valids) },
 		},
+		{
+			logical:                          "text",
+			physical:                         "string with invalid utf8",
+			sc:                               arrow.NewSchema([]arrow.Field{{Type: &arrow.StringType{}}}, nil),
+			rowType:                          execResponseRowType{Type: "TEXT"},
+			values:                           []string{"\xFF", "bar"},
+			enableArrowBatchesUtf8Validation: true,
+			nrows:                            2,
+			builder:                          array.NewStringBuilder(pool),
+			append:                           func(b array.Builder, vs interface{}) { b.(*array.StringBuilder).AppendValues(vs.([]string), valids) },
+			compare: func(src interface{}, convertedRec arrow.Record) int {
+				srcvs := src.([]string)
+				arr := convertedRec.Column(0).(*array.String)
+				for i := 0; i < arr.Len(); i++ {
+					if !utf8.ValidString(arr.Value(i)) || strings.ToValidUTF8(srcvs[i], "�") != string(arr.Value(i)) {
```

Comment:
Perhaps we could somehow check converted values against literals, to avoid reusing `strings.ToValidUTF8(srcvs[i], "�")` which is our production code?

Reply:
make sense, changed.